### PR TITLE
Further serial gimbal/headtracker improvements

### PR DESCRIPTION
**Pull request by mmosca**
A follow up to #10109, for further improvement.

* [x] Add trim to gimbal to allow Uptilt on quad setups. This only applies to locked center mode.
```
set gimbal_pan_trim = 0
set gimbal_tilt_trim = 0
set gimbal_roll_trim = 0
```

* [x] Integrate with OSD pan compensation

### CODE REVIEW

**Comment by mmosca** [comment]
@MrD-RC  can you check my changes to osd.c:osdGetPanServoOffset() ? This should get the pan compensation working, but I would like to double check if my logic for choosing servo input vs serial gimbal pan is sound.

**Comment by MrD-RC** [comment]
> @MrD-RC  can you check my changes to osd.c:osdGetPanServoOffset() ? This should get the pan compensation working, but I would like to double check if my logic for choosing servo input vs serial gimbal pan is sound.

It looks good to me 👍🏻

File: src/main/io/osd.c [hunk 1]
```diff
@@ -1203,8 +1204,15 @@ int16_t osdGetHeading(void)
...
 {
     int8_t servoIndex = osdConfig()->pan_servo_index;
-    int16_t servoPosition = servo[servoIndex];
     int16_t servoMiddle = servoParams(servoIndex)->middle;
+    int16_t servoPosition = servo[servoIndex];
+
+    gimbalDevice_t *dev = gimbalCommonDevice();
+    if (dev && gimbalCommonIsReady(dev)) {
+        servoPosition = gimbalCommonGetPanPwm(dev);
+        servoMiddle = PWM_RANGE_MIDDLE;
```
Comment (MrD-RC) [inline src/main/io/osd.c]: One question. Shouldn't this use gimbal_pan_trim for the central position?

Comment (mmosca) [reply src/main/io/osd.c]: Makes sense. I updated everything to use trim, (except, raw headtracker).

If no valid headtracker is found, behavior is the same as "center lock" as well.

If there is pan trim, osd will reflect what happens for non head tracked modes as well, so you may need to look a bit sideways to center it in osd. (looking the same direction as non head tracked mode).

**Pull request merged**